src/app/pages/portal/portal.component.ts

```diff
@@ -490,14 +522,102 @@ export class PortalComponent implements OnInit, OnDestroy {
 
   ngOnDestroy() {
     this.context$$.unsubscribe();
-    this.activeWidget$$.unsubscribe();
-    this.openSidenav$$.unsubscribe();
     this.workspaceMaximize$$.map(f => f.unsubscribe());
-    this.sidenavMediaAndOrientation$$.unsubscribe();
+    this.layers$$?.unsubscribe();
+    this.searchTerm$$.unsubscribe();
   }
 
-  removeFeatureFromMap() {
-    this.map.searchResultsOverlay.clear();
+  private getQuerySearchSource(): SearchSource {
+    return this.searchSourceService
+      .getSources()
+      .find(
+        (searchSource: SearchSource) =>
+          searchSource instanceof QuerySearchSource
+      );
+  }
+
+  private getFeatureIsSameActiveWks(feature: Feature): boolean {
+    if (this.workspace) {
+      const featureTitle = feature.meta.sourceTitle;
+      const wksTitle = this.workspace.title;
+      if (wksTitle === featureTitle) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+    return false;
+  }
+
+  private getWksActiveOpenInResolution(): boolean {
+    if(this.workspace) {
+      const activeWks = this.workspace as WfsWorkspace;
+      if(activeWks.active && activeWks.inResolutionRange$.value && this.workspaceState.workspacePanelExpanded) {
+        return true;
+      }
+    }
+    return false;
+   }
+
+  onMapQuery(event: { features: Feature[]; event: MapBrowserEvent<any> }) {
+    if(this.configService.getConfig('queryOnlyOne')){
+      event.features = [event.features[0]];
+      this.map.queryResultsOverlay.clear(); // to avoid double-selection in the map
+    }
+    const baseQuerySearchSource = this.getQuerySearchSource();
+    const querySearchSourceArray: QuerySearchSource[] = [];
+    if (event.features) {
```

Comment:
Après avoir tester, on devrait valider que la liste n'est pas vide
```suggestion
    if (event.features?.length) {
```